Automattic/jetpack
# Social | Migrate Social plugin settings to the new script-data

## Proposed changes:

* Migrate Social plugin settings to the new script-data
* To keep the changes in this PR to be minimal, I did the clean up in a separate PR - #40033

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Follow these instructions checking out #40033 to ensure both PRs get tested.
* Go to social admin page
* Confirm that you see the publicize and social notes toggle
* Turn them ON and OFF
* Confirm that they work fine
* Confirm that changes are persisted after page reload
* Change social notes options
* Confirm that they are persisted as before
* Confirm that there is no page reload when disabling Social Notes
* Confirm that the connections management is disabled (inert) while social notes are updated.
* Do a sanity check for Jetpack settings page and the editor
* On a fresh JN site, confirm that the pricing page is shown on social admin page the first time
* Click on Start for free
* Confirm that the pricing table is hidden
* Reload the page
* Confirm that the change is persisted
